60_networks.Rmd

```diff
@@ -0,0 +1,855 @@
+       bty = "n", horiz = TRUE)
+
+# Colors used in the legend should be equally transparent as in the plot
+phylcol_transp <- colToTransp(phylcol, 60)
+
+legend(-1.8, 1.1, cex = 1.7, pt.cex = 2.5, title = "Phylum:", 
+       legend=levels(phyla), col = phylcol_transp, bty = "n", pch = 16) 
+```
+
+**A few things to observe:**
+
+-   Genera belonging to the same phylum tend to cluster together, though not perfectly.
+-   Genera with a low total count play a rather unimportant role in the network, i.e., they have a low centrality.
+-   There is only one negative edge in the network. This edge is between two clusters, as expected when using the "signed" transformation.
+
+## Differential network analysis
+
+In addition to the count matrix, the data set also includes information on the subjects' geographical location, gender, age, and diet.
+
+An interesting question to explore now is whether the skin microbiome differs between people whose diets are different.
+
+```{r sample_data}
+table(colData(tse_genus)$Diet)
+```
+
+We split the data set into two phyloseq objects: One with mixed diet subjects and one with vegetarian subjects. Both subsets have nearly the same sample size and are therefore comparable.
+
+```{r split_data}
+tse_mixed <- tse[, colData(tse_genus)$Diet == "Mixed"]
+tse_veg <- tse[, colData(tse_genus)$Diet == "Veg"]
+```
+
+### Network construction
+
+The approach starts again with network construction and analysis, but this time we pass the two data sets to `netConstruct` to perform a network comparison.
+
+The `rep.num` argument is set to 10 to perform only 10 repetitions in the model selection approach. This speeds up the permutation tests performed later, and has a negligible effect for this data set.
+
+```{r netConstruct_diet, eval=FALSE}
+spring_net_diet <- netConstruct(data = tse_mixed,
+                                data2 = tse_veg,
+                                taxRank = "genus",
+                                filtTax = "highestFreq",
+                                filtTaxPar = list(highestFreq  = 100),
+                                measure = "spring",
+                                measurePar = list(nlambda = 20, 
+                                                  rep.num = 10,
+                                                  thresh = 0.05,
+                                                  Rmethod = "approx"),
+                                sparsMethod = "none", 
+                                dissFunc = "signed",
+                                verbose = 3,
+                                seed = 13075)
+```
+
+### Network analysis
+
+All network measures are now computed for both networks. Also, both GCMs are plotted together with a third matrix containing the differences between the GCMs and significance codes that express if the differences are significantly different from zero.
+
+```{r netAnalyze_diet, fig.width=10, fig.height=10}
```

Comment:
There are many subsections; consider if some of the material could be included in a supplement that is only linked from here, to keep the main text more compact. Optional, for consideration.

Reply:
Thanks for the suggestions! Steffi and I discussed this and we would suggest to split  the chapter into two chapters. The first chapter contains single network estimation and analysis. The second chapter  will be about Differential Network analysis and will comprise 15.6 and 15.7.  The second chapter will then mirror Chapter 11 with DAA and use instead Differential Network Analysis ("DNA"). Like this, the reader may start with the first chapter as intro, and the second chapter assumes, one already knows about network learning...

Reply:
Hi! Apologies for the delays due to travel. This sounds good to me!